src/main/java/dev/mcnees/moneymapper/MoneyMapperRunner.java

```diff
@@ -0,0 +1,97 @@
+package dev.mcnees.moneymapper;
+
+import java.io.File;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.sql.DataSource;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.batch.core.Job;
+import org.springframework.batch.core.JobParameter;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.JobParametersBuilder;
+import org.springframework.batch.core.launch.JobLauncher;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MoneyMapperRunner implements ApplicationRunner {
+
+	private static final Logger log = LoggerFactory.getLogger(MoneyMapperRunner.class);
+
+	private final JobLauncher jobLauncher;
+
+	private final Job moneyMapperJob;
+
+	private final JdbcTemplate jdbcTemplate;
+
+	@Value("${qfx_directory:qfx_files}")
+	private String qfx_directory_parameter;
+
+	@Value("${output_directory:output}")
+	private String output_directory_parameter;
+
+	public MoneyMapperRunner(JobLauncher jobLauncher, Job moneyMapperJob, DataSource dataSource) {
+		this.jobLauncher = jobLauncher;
+		this.moneyMapperJob = moneyMapperJob;
+		jdbcTemplate = new JdbcTemplate(dataSource);
+	}
+
+	@Override
+	public void run(ApplicationArguments args) throws Exception {
+
+		clearProcessingData();
+
+		File qfx_directory = new File(qfx_directory_parameter);
```

Comment:
I tried to use a [Resource](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/io/AbstractResource.html), specifically a [FileSystemResource](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/core/io/FileSystemResource.html), instead but I can't find a good replacement in the API for the [File isDirectory()](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/File.html#isDirectory%28%29) or [File listFiles](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/io/File.html#listFiles%28%29) method.

It seems to me like the Resource API is optimized for files, not directories.  Since I need to initially work with directories I have them as File objects first.  I made a change to convert them to a Resource before passing to JobParameters.

Reply:
I think all of this changes if you use the `MultiResourceItemReader`. Then this whole class should go away anyways.

Reply:
OK I'll evaluate it in the context that change.

You didn't call out the variable names but I noticed that I switched to python conventions for some reason.  I changed the variable names.